vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/per_hour_frequency_cap.cc

```diff
@@ -0,0 +1,40 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/frequency_capping/exclusion_rules/per_hour_frequency_cap.h"
+#include "bat/ads/internal/frequency_capping/frequency_capping.h"
+#include "bat/ads/internal/time.h"
+#include "bat/ads/internal/client.h"
+
+#include "bat/ads/ad_info.h"
+
+namespace ads {
+
+bool PerHourFrequencyCap::ShouldExclude(
+  const AdInfo& ad) {
+  if (!DoesAdRespectPerHourCap(ad)) {
+    std::ostringstream string_stream;
```

Comment:
nit: You could also use `base::StringPrintf("Foobar is set to %d", foobar)`. `base` provides lots of helpers, see https://cs.chromium.org/search/?type=cs

Reply:
Yeh. I'll probably stick with StringPrintf in future - but will leave as ostringstream for this PR.